tests/test_decay_angles_dialog.py

```diff
@@ -1,11 +1,22 @@
+import pytest
 from pytestqt.qtbot import QtBot
 
-from cavendish_particle_tracks._decay_angles_dialog import DecayAnglesDialog
 
-
-def test_decay_angles_dialog(cpt_widget, qtbot: QtBot):
+@pytest.mark.parametrize(
+    "double_click",
+    [
+        True,
+        False,
+    ],
+)
+def test_decay_angles_dialog(cpt_widget, qtbot: QtBot, double_click):
     """Smoke test for the DecayAnglesDialog."""
-    dialog = DecayAnglesDialog(parent=cpt_widget)
+    cpt_widget.cb.setCurrentIndex(4)
+
+    dialog = cpt_widget._on_click_decay_angles()
+    if double_click:
+        dialog = cpt_widget._on_click_decay_angles()
```

Comment:
I would reverse this, if clicking twice then call the `_on_click...` function without catching the return:
```suggestion
    # The user clicks the button (also test fringe case that they click the button again).
    if click_twice:
        cpt_widget._on_click_decay_angles()
    dialog = cpt_widget._on_click_decay_angles()
```